dockerfiles/theia/Dockerfile

```diff
@@ -11,54 +11,62 @@
 ###
 # Builder Image
 #
-FROM node:8.12.0-alpine as builder
-
-ARG GITHUB_TOKEN
-
-# Install basic software used for checking github API rate limit
-RUN apk add --update --no-cache curl jq expect
+FROM ${BUILD_ORGANIZATION}/${BUILD_PREFIX}-theia-dev:${BUILD_TAG} as builder
+WORKDIR ${HOME}
 
 # define in env variable GITHUB_TOKEN only if it is defined
 # else check if github rate limit is enough, else will abort requiring to set GITHUB_TOKEN value
-ADD src/builder/check-github-limit.sh /tmp/builder/scripts/check-github-limit.sh
-RUN /tmp/builder/scripts/check-github-limit.sh
-
-# Install tools required to build theia
-RUN apk add make gcc g++ python git bash
+ARG GITHUB_TOKEN
 
 # Define upstream version of theia to use
 ARG THEIA_VERSION=0.3.17
 
-ENV THEIA_VERSION=${THEIA_VERSION} \
-    HOME=/home/theia
-WORKDIR ${HOME}
-
-# add eclipse che theia generator
-RUN yarn global add @eclipse-che/theia-generator@0.0.1-1543568125
-
+# Invalidate cache if there is a new commit
+ADD https://api.github.com/repos/theia-ide/theia/git/refs/tags/v${THEIA_VERSION} /tmp/.sha1-repository
+
+# Check github limit
+RUN if [ ! -z "${GITHUB_TOKEN-}" ]; then \
+      export GITHUB_TOKEN=$GITHUB_TOKEN; \
+      echo "Setting GITHUB_TOKEN value as provided"; \
+    else \
+      export GITHUB_LIMIT=$(curl -s 'https://api.github.com/rate_limit' | jq '.rate .remaining'); \
+      echo "Current API rate limit https://api.github.com is ${GITHUB_LIMIT}"; \
+      if [ "${GITHUB_LIMIT}" -lt 10 ]; then \
+        printf "\033[0;31m\n\n\nRate limit on https://api.github.com is reached so in order to build this image, "; \
+        printf "the build argument GITHUB_TOKEN needs to be provided so build will not fail.\n\n\n\033[0m"; \
+        exit 1; \
+      else \
+        echo "GITHUB_TOKEN variable is not set but https://api.github.com rate limit has enough slots"; \
+      fi \
+    fi
+
+# Clone theia
+RUN git clone --branch v${THEIA_VERSION}  --single-branch --depth 1 https://github.com/theia-ide/theia ${HOME}/theia-source-code
 
 # Add patches
-ADD src/patches /home/theia-build/patches
+ADD src/patches ${HOME}/patches
 
-# Invalidate cache if there is a new commit
-ADD https://api.github.com/repos/theia-ide/theia/git/refs/tags/v${THEIA_VERSION} /tmp/sha1-repository
-# Build theia from source code and publish artifacts locally
-ADD src/builder/clone.sh /tmp/builder/scripts/clone.sh
-RUN /tmp/builder/scripts/clone.sh
+# Apply patches
+RUN if [ -d "${HOME}/patches/${THEIA_VERSION}" ]; then \
+      echo "Applying patches for Theia version ${THEIA_VERSION}"; \
+      for file in $(find "${HOME}/patches/${THEIA_VERSION}" -name '*.patch'); do \
+        echo "Patching with ${file}"; \
+        cd ${HOME}/theia-source-code && patch -p1 < ${file}; \
+      done \
+    fi
 
 # Generate che-theia
-RUN cd ${HOME}/theia-source-code && che:theia init
+WORKDIR ${HOME}/theia-source-code
+RUN che:theia init
```

Comment:
Doesn't it clone Theia?

Reply:
no, currently init  doesn't clone. It only configures directories / clone extensions